### ISSUE TITLE
[Feature] Show combined tracked time across the project by company or organization name

### ISSUE THREAD

**Issue by pparh602** [Swing-team/pendulums-web-client#126]
Based on the fact that the majority of Pendulums users are developers they are working on different project. For example I'm freelancer but I'm working for company which has multiple different application. I'm tracking the time separately by creating new Project. Currently, we can see the stats by project, but not the combined tracked work or by company. So, IMO we should add one more stats such as by company name. And then show the states so if one is working on multiple project for the same company.

**Comment by mohammadrafigh** [comment]
Great suggestion, Thank you. We are adding a new chart in the dashboard to show an overview of the user in all projects, but we can also let users choose between a selectable list of projects which will do what you're looking for. But generally adding some categorization/organization/folder is a cool idea.

**Comment by pparh602** [comment]
Additionally, I propose a new feature for the desktop application. In instances where a user forgets to manually stop the timer, the application should offer an automatic timer stop functionality accompanied by an email reminder. Presently, users receive email reminders indicating that their timer is still running. However, to optimize efficiency, the application should automatically halt the timer after a designated period of inactivity, typically 15 minutes, if it detects no user interaction. Alternatively, users could be presented with the option to decide whether the time spent offline should be included in the tracking or deducted upon resuming activity within the application.

**Comment by pparh602** [comment]
Also, I'm interested in contributing to the app! Could you point me towards any available developer documentation? Additionally, I was wondering if it would be beneficial to create various ER diagrams to facilitate smoother development.

**Comment by mohammadrafigh** [comment]
> Additionally, I propose a new feature for the desktop application. In instances where a user forgets to manually stop the timer, the application should offer an automatic timer stop functionality accompanied by an email reminder. Presently, users receive email reminders indicating that their timer is still running. However, to optimize efficiency, the application should automatically halt the timer after a designated period of inactivity, typically 15 minutes, if it detects no user interaction. Alternatively, users could be presented with the option to decide whether the time spent offline should be included in the tracking or deducted upon resuming activity within the application.

Yes we had a talk to do it automatically, and we may add it to Pendulums 2.0, but having an email action would be interesting too, thanks for sharing your thoughts.

> Also, I'm interested in contributing to the app! Could you point me towards any available developer documentation? Additionally, I was wondering if it would be beneficial to create various ER diagrams to facilitate smoother development.

Great! I will add a doc soon. But for now:

1. Clone the repository
2. Do a `npm install` (Please use node 10, 12, 14. We are in the process of upgrading node.)
3. Open `prepare.js` file and edit these properties like this:

```
  ...
  socketEndpoint: 'https://alpha.pendulums.io/api',
  socketPath: '/api/socket.io',
  apiEndpoint: 'https://alpha.pendulums.io/api',
  httpOptions: { withCredentials: true, responseType: 'json' },
  imagesEndpoint: 'https://alpha.pendulums.io/media/images',
  ...
```
4. Run the project using `npm run start`.

Regarding the backend, we are working seriously to open source it ASAP so you can self-host it or use it easily in development. But for now, please use our `alpha.pendulums.io` as the backend in development.

Regarding the diagrams, we already have some internal ER and UML diagrams implemented for the backend and will release them when open sourcing the backend. But as I know, there is nothing useful for the client. So it could be useful if you care.

We are currently working on Pendulums 2.0. So please contact us using the contact form on the website or join into this matrix group: [#pendulums:matrix.swing.team](https://matrix.to/#/#pendulums:matrix.swing.team) so we can stay synced in what features or issues that we appreciate you contribute in.

Feel free to ask any questions, thanks in advance.

**Comment by billybhaub** [comment]
Hi. I just want to show my support for a more comprehensive way to see the time I worked on many different projects.

Here's my situation, to help you understand my needs.

I'm a self-employed entrepreneur. I work on multiple projects at the same time (say, a conference, a book, seeking contracts, bookkeeping, and much, much more!). So, I really like how easy Pendulums makes tracking time on all those different projects.

What I would like is the ability, at the end of the week (or month, or year!) to have a chart showing me how much time I spent on each projects, _and_ how much time I spent on my business overall.

This is the only thing that I see that is missing on Pendulums, which is otherwise exactly what I'm looking for as a time tracker : simple, intuitive, no useless bells and whistles.

Thank you.
